ERCS/erc-7579.md

```diff
@@ -48,149 +48,164 @@ The smart account's `validateUserOp` function SHOULD return the return value of
+- execType (1 byte): `0x00` for a normal execution, `0x01` for a try/catch execution
+- unused (4 bytes): this range is reserved for future standardization
+- modeSelector (4 bytes): an additional mode selector that can be used to create further execution modes
+- modePayload (22 bytes): additional data to be passed
+
+Here is a visual representation of the execution mode:
+
+| CallType | ExecType | Unused  | ModeSelector | ModePayload |
+| -------- | -------- | ------- | ------------ | ----------- |
+| 1 byte   | 1 byte   | 4 bytes | 4 bytes      | 22 bytes    |
+
+#### Account configurations
+
+To comply with this standard, smart accounts MUST implement the account config interface below:
 
+```solidity
+interface IAccountConfig {
     /**
-    * MUST execute a `call` to the target with the provided data and value
-    * MUST only allow installed executors to call this function
-    * MUST revert if the call was not successful
-    */
-    function executeFromExecutor(address target, uint256 value, bytes callData) external returns (bytes memory result);
+     * @dev Initializes the account. Function might be called directly or by a factory
+     * @param data encoded data that can be used during the initialization phase
+     *
+     * MUST revert if the account is already initialized
+     */
+    function initializeAccount(bytes calldata data) external payable;
 
     /**
-    * MUST execute a `call` to all the targets with the provided data and value for each target
-    * MUST only allow installed executors to call this function
-    * MUST revert if any call was not successful
-    */
-    function executeBatchFromExecutor(Execution[] calldata executions) external returns (bytes memory result);
-}
+     * @dev Returns the account id of the smart account
+     * @return accountImplementationId the account id of the smart account
+     *
+     * MUST return a non-empty string
+     * The accountId SHOULD be structured like so:
+     *        "vendorname.accountname.semver"
+     * The id SHOULD be unique across all smart accounts
+     */
+    function accountId() external view returns (string memory accountImplementationId);
 
-/**
-* @dev implementing delegatecall execution on a smart account must be considered carefully and is not recommended in most cases
-*/
-interface IExecutionUnsafe {
     /**
-    * MUST execute a `delegatecall` to the target with the provided data
-    * MUST allow ERC-4337 Entrypoint to be the sender and MAY allow `msg.sender == address(this)`
-    * MUST revert if the call was not successful
-    */
-    function executeDelegateCall(address target, bytes data) external returns (bytes memory result);
+     * @dev Function to check if the account supports a certain CallType or ExecType
+     * @param encodedMode the encoded mode
+     *
+     * MUST return true if the account supports the mode and false otherwise
+     */
+    function supportsAccountMode(bytes32 encodedMode) external view returns (bool);
```

Comment:
Also mention **modeSelector** and **modePayload**?

Reply:
yep this will be part of the encoded mode - will make this clear